File: BackEnd_Learn/Java/awss3/src/main/java/com/leeseojune/awss3/error/GlobalExceptionHandler.java
package com.leeseojune.awss3.error;

import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.ControllerAdvice;
import org.springframework.web.bind.annotation.ExceptionHandler;
import org.springframework.web.multipart.MultipartException;

@ControllerAdvice
public class GlobalExceptionHandler {

//    @ExceptionHandler(MultipartException.class)
//    protected ResponseEntity<ErrorResponse> testHandler(MultipartException e) {
//        return new ResponseEntity<>(new ErrorResponse(200, "No File"), HttpStatus.OK);
//    }

}

File: BackEnd_Learn/kotlin/kopring/src/main/kotlin/io/github/leeseojune53/kopring/global/error/GlobalExceptionHandler.kt
package io.github.leeseojune53.kopring.global.error

import io.github.leeseojune53.kopring.global.error.exception.ErrorCode
import io.github.leeseojune53.kopring.global.error.exception.KopringException
import org.springframework.http.HttpStatus
import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.ExceptionHandler
import org.springframework.web.bind.annotation.RestControllerAdvice

@RestControllerAdvice
class GlobalExceptionHandler {

    @ExceptionHandler(KopringException::class)
    fun handlingKopringException(e: KopringException): ResponseEntity<ErrorResponse> {
        return ResponseEntity(ErrorResponse(e.status, e.errorMessage), HttpStatus.valueOf(e.status))
    }

}